src/queryIterator.ts

```diff
@@ -64,7 +67,7 @@ export class QueryIterator<T> {
     this.reset();
     let index = 0;
     while (this.queryExecutionContext.hasMoreResults()) {
-      const result = await this.queryExecutionContext.nextItem();
+      const result = await this.executeMethod("nextItem");
```

Comment:
Isn't this a bit fragile? what if someone changes the name of the method? And does it come with intellisense support?

Reply:
It does work fully with intellisense. The type below `Exclude<keyof IExecutionContext, "hasMoreResults">` gives us pretty good checking.  I'll noodle on if there is a better way to do this.